src/expression.c

```diff
@@ -10015,12 +10086,12 @@ Expression *SliceExp::semantic(Scope *sc)
         }
         else if (t1b->ty == Tpointer)
         {
-            this->upperIsInBounds = true;
+            this->upperIsInBounds = true; // no boundscheck for x.ptr[_ .. _]
         }
         else
             assert(0);
 
-        this->lowerIsLessThanUpper = (lwrRange.imax <= uprRange.imin);
+        this->lowerIsLessThanUpper = this->lowerIsLessThanUpper || (lwrRange.imax <= uprRange.imin);
```

Comment:
I would use `if(lwrRange.imax <= uprRange.imin) this->lowerIsLessThanUpper = true;` here.